### ISSUE TITLE
Individual cookies can't be accessed properly in some domains through `brave://settings/content/all` when "subdomains" exist.

### ISSUE THREAD

**Issue by ghost** [brave/brave-browser#31690]
## Description 

@spylogsster 
I found I found there is an issue seeing the individual cookies (from https://github.com/brave/brave-core/pull/17746 - `Added back Chromium implementation of views to look for individual cookies stored per-website`) in Brave `brave://settings/content/all` for many domains where subdomains exist or where domains storages are done in `www.example.com` vs `youtube.com` like Youtube, where cookies are stored in `youtube.com` and the other storages are stored in `www.youtube.com`

Take a look at this screenshot:
![image](https://github.com/brave/brave-browser/assets/122518587/c69667b7-182a-4eb8-a645-e20497a36ec5)

As you can see there is no `>` right arrow in `youtube.com`, because it was replaced by the up and down arrow to expand and collapse domains, and only in `www.youtube.com`

and since there is no arrow to go to `youtube.com` and only to expand per 'subdomains', permissions or `Cookies and Site Data` button, can't be accessed unless the URL is manually changed to remove the `www.` part of it: `brave://settings/content/siteDetails?site=https%3A%2F%2Fwww.youtube.com%2F` to `brave://settings/content/siteDetails?site=https%3A%2F%2Fyoutube.com%2F`

Once done that, then the Cookies and Site Data button will go to the correct place. 
![image](https://github.com/brave/brave-browser/assets/122518587/eef8822e-c229-4ff3-99a1-bd68a9fe6b2a)

For example, this is what you get when you go to `www.youtube.com` one.
![image](https://github.com/brave/brave-browser/assets/122518587/aefe9c1c-7c3b-4fab-818b-6c078d69a130), so it doesn't include the cookies since they were added in the `youtube.com` one, and for Browser's logic, they are different things, even the permissions are set differently, because one will be seen as domain and the other as subdomain I assume?

Anyway, I tested in a complete new User Data to avoid any issues with flags and anything else and same behavior, I also tested in Stable version and same issue where Expanding/Collapse arrow will replace it.

I must admit I rarely use this feature, but today I was testing Youtube and I saw I had to manually edit the URL to get the cookies, so not a big deal for me, but Brave team might want to fix the issue since it can mess with many domains, like `live.com` is another example, where you can't really access the `live.com` information, so you don't get to see all the cookies 32 cookies Brave says it has, and you can only access 24.

![image](https://github.com/brave/brave-browser/assets/122518587/893e8ecc-44eb-4cbb-be1e-0348fd6b85d7)

Missing all these 8 cookies, only because `live.com` can't be accessed once a subdomain takes place.

![image](https://github.com/brave/brave-browser/assets/122518587/8465942a-7d59-40d9-8c34-6115576d28db)

Thank you and have a good day.

## Steps to Reproduce

   1. Go to youtube
   2. then go to brave://settings/content/all
   3. only `www.youtube.com` data can be accessed and not `youtube.com`

## Brave version (brave://version info)

1.58.21 Chromium: 115.0.5790.75 (Official Build) nightly (64-bit) 

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the nightly channel? Yes

**Comment by spylogsster** [comment]
cc @rebron

**Comment by ShivanKaul** [comment]
Yeah, it looks like we don't show the cookies set for the top-level Domain. In the example in the issue description, you can see the cookies in dev console; they have their Domain set to be `.youtube.com`. We should show these under youtube.com.
